### ISSUE TITLE
Highlighter Tool

### ISSUE THREAD

**Issue by babluboy** [babluboy/bookworm#102]
Add a tool to highlight text in a book. Highlighted text should be preserved and also removed when required

Perhaps use fluorescent green for highlighting to make it different from searched text which could be highlighted in fluorescent yellow

**Comment by baimafeima** [comment]
Out of curiosity, would all the extra information such as highlighting and underlining be stored in another file? If yes, how would you ensure compatibility to other major applications such as Okular? Ideally, there should be no software lock-down, that is, that annotations made with one programme should be readable and editable with another.

**Comment by babluboy** [comment]
@baimafeima That is a good thought to have cross software support. However, I could not find any standards for eBook meta data to which all eBook software can align. So will have a Bookworm specific installation for highlighted text (data will be stored in the Bookworm sqllite database)....

maybe a future thought can be to align to some agreed standard to ensure compatibility with other eBook software....

**Comment by tonypachino** [comment]
Bookworm already looks like iBooks on Mac, can you make same highlight feature like there in iBooks, that we can see all highlighted texts on the left side, separated, highlight by highlight and when we click on highlight, we jump to that highlighted text in the book?

**Comment by babluboy** [comment]
@tonypachino thanks, that sounds useful. Will be great if you can post a iBook screen shot here so I will have a better idea of the UX when I come around to implementing this.

**Comment by tonypachino** [comment]
@babluboy you can see all this here in this link https://www.youtube.com/watch?v=pSLE6UP0eSA

I think that highlighting and making notes is the most important thing when people reading books

**Comment by babluboy** [comment]
Many thanks for the pointer...I'm also planning to introduce a collapsible sidebar. The side bar can be the container for filtering and for highlighted annotation

**Comment by babluboy** [comment]
I have completed highlighting and annotation. The implementation is not exactly like iBooks but hopefully it should be easy to accomplish without a side bar:
1- Search results within the book are highlighted and the page is auto scrolled to the highlighted text
2- Selected text can be right clicked to annotate and will appear highlighted to easily identify annotations. Annotated text can be clicked to edit the annotation as well.
3- Annotations within the book are listed and clicking on a annotation will jump to the page where the annotation is present
4- Annotations can be tagged with key words and these key words can be used in the library to filter books as well.

Closing this as the feature was tracked by #109
